src/zope/interface/interface.py

```diff
@@ -591,7 +595,7 @@ def __hash__(self):
         try:
             return self._v_cached_hash
         except AttributeError:
-            self._v_cached_hash = hash((self.__name__, self.__module__))
+            self._v_cached_hash = hash((self.__qualname__, self.__module__))
```

Comment:
I'm afraid that changing this here means that `__cmp` must also change to use `__qualname__`. As it is, right now, I think the [Python language requirement](https://docs.python.org/3/reference/datamodel.html#object.__hash__) that "objects that compare equal must have equal hash codes" is broken.

Because of that, I'm also afraid that the apparent fix could be due to misbehaving dictionaries.